Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Internal/CoreRemoteMongoClient.swift

```diff
@@ -27,4 +33,36 @@ public class CoreRemoteMongoClient {
                                             withService: service,
                                             withDataSynchronizer: dataSynchronizer)
     }
+
+    private func onAuthEvent(_ authEvent: AuthRebindEvent) {
+        switch authEvent {
+        case .userRemoved(let removedUser):
+            let userId = removedUser.id
+            let key = appInfo.clientAppID + "/\(userId)"
+            try? CoreLocalMongoDBService.shared.deleteDatabase(
+                withKey: key,
+                withDBPath: "\(appInfo.dataDirectory.path)/local_mongodb/0/\(key)")
+        case .activeUserChanged(let currentActiveUser, _):
+            if lastActiveUserId != appInfo.authMonitor.activeUserId {
```

Comment:
Seems strange that you are not just getting the new authInfo from currentActiveUser, but I guess it shouldnt be a problem?

Reply:
`currentActiveUser` is not available here, as this is a service client, not the `StitchAuth`. the `authMonitor` serves as the reference to auth information for this service client

Reply:
I meant the variable passed in to the method

Reply:
I didn't design this so I can't speak to it in depth, but from what I can tell, it's about keeping ` CoreRemoteMongoClient` simple, and not giving it too much state to maintain. 

The appInfo maintains a reference to the app client and auth but as an `AuthMonitor` rather than a full-fledged app client. This gives the CoreRemoteMongoClient only the state it needs to do its job, and nothing more.

@jsflax can probably speak more to this when he's back

Reply:
Correct.